Build/SaveLoadGame.cc

```diff
@@ -1466,6 +1479,33 @@ static void LoadSoldierStructure(HWFILE const f, UINT32 savegame_version, bool s
 static void WriteTempFileNameToFile(const char* pFileName, UINT32 uiSizeOfFile, HWFILE hSaveFile);
 #endif
 
+void BackupSavedGame(UINT8 const ubSaveGameID)
+{
+	std::string backupdir = FileMan::joinPaths(GCM->getSavedGamesFolder().c_str(),"Backup");
+	FileMan::createDir(backupdir.c_str());
+	char zSourceSaveGameName[512];
+	char zSourceBackupSaveGameName[512];
+	char zTargetSaveGameName[512];
+	sprintf(zSourceSaveGameName, "%s%02d.%s", g_savegame_name, ubSaveGameID, g_savegame_ext);
+	for (int i = NUM_SAVE_GAME_BACKUPS - 1; i >= 0; i--)
+	{
+		if (i==0)
+		{
+			strcpy(zSourceBackupSaveGameName, zSourceSaveGameName);
+		}
+		else
+		{
+			sprintf(zSourceBackupSaveGameName, "%s.%01d", zSourceSaveGameName, i);
```

Comment:
ok. Since truncating the name is a bad idea, the target buffer has to be large enough to accommodate all the extra chars for the extension. So just increase that 512 and make sure the string is null terminated.